sklearn/model_selection/_search.py

```diff
@@ -840,8 +841,12 @@ def evaluate_candidates(candidate_params, cv=None, more_results=None):
                         )
                     )
 
+                # Capture the config of the current thread here instead of inside the
+                # generator expression. The generator expression can be consumed by
+                # an auxiliary thread in joblib.
```

Comment:
This comment was written only once for the purpose of this PR only. Repeating it everywhere might be too verbose. Not sure what to do...

Maybe the content of the warning message is explicit enough.